frontend/src/components/CheckboxGroup/CheckboxGroupValidator.tsx

```diff
@@ -0,0 +1,221 @@
+import { List } from 'immutable';
+import React, { FunctionComponent, useState } from 'react';
+import { Filters, OperationStepMap } from '../../types/operations';
+import { BasicModal } from '../BasicModal';
+
+interface ComponentProps {
+  steps?: List<OperationStepMap>;
+}
+
+const CheckboxGroupValidator: FunctionComponent<ComponentProps> = (props) => {
+  const [showModal, setShowModal] = useState<boolean>(false);
+  const [modalMessage, setModalMessage] = useState<string>('');
+
+  const validateCheckboxOptions = (
+    checkboxValue: string,
+    checkboxState: boolean | undefined,
+  ): void => {
+    const { steps } = props;
+    steps &&
+      steps.map((step) => {
+        const options = step.get('query_kwargs') as string;
+        toggleModalState(options, checkboxState, checkboxValue);
+      });
+  };
+
+  const toggleModalState = (
```

Comment:
Rename this `validateColumnDeselect`

Reply:
Also, as far I can tell, the only code related to the component that this function uses are state updates. That makes it a good candidate for extraction into a `utils` file (under the same component folder)`. Make it a promise.
i.e. What I expect to see here is:

`validateColumnDeselect(...).then(validationMessage => if (validationMessage) { setModalMessage(validationMessage); }`

Reply:
Moved to `utils` and updated to return a promise